code/modules/reagents/reagents/dispenser.dm

```diff
@@ -430,21 +430,19 @@
 	glass_name = "sugar"
 	glass_desc = "The organic compound commonly known as table sugar and sometimes called saccharose. This white, odorless, crystalline powder has a pleasing, sweet taste."
 
-/datum/reagent/organic/sugar/affect_blood(mob/living/carbon/M, alien, effect_multiplier)
-	M.adjustNutrition(1 * effect_multiplier)
```

Comment:
Pretty sure this will affect all food that has sugar even when not injected into blood.